src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,58 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+
 public class SalaryInfo {
+
+    private static final int WORK_DAY = 0;
+    private static final int HOURS = 2;
+    private static final int INCOME_PER_HOUR = 3;
+    private static final int WORKER_NAME = 1;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder info = new StringBuilder("Report for period ");
+        info.append(dateFrom).append(" - ").append(dateTo);
+        SalaryInfo salaryInfo = new SalaryInfo();
+        LocalDate fromDate = salaryInfo.toLocalDate(dateFrom);
+        LocalDate toDate = salaryInfo.toLocalDate(dateTo);
+        String[] splitedData;
+        LocalDate workDayDate;
+        int salary;
+        for (String name : names) {
+            salary = WORK_DAY;
```

Comment:
it is ok to initialize it with plain `0` and not a constant
```suggestion
        int salary = 0;
        for (String name : names) {
```
regarding the constants usage we can and should reuse them, but only if the context allows us, here work day index of the data array has nothing to do with the salary var initialization